src/app/auth/authentication.service.ts

```diff
@@ -2,12 +2,13 @@ import { Injectable, Inject } from '@angular/core';
 import { Http, Response, Headers, RequestOptions, RequestOptionsArgs } from '@angular/http';
 
 import { Observable, Subject } from 'rxjs';
-import { Broadcaster } from 'ngx-base';
+import { Broadcaster, Logger } from 'ngx-base';
 
 import { AUTH_API_URL } from '../shared/auth-api';
 import { SSO_API_URL } from '../shared/sso-api';
 import { REALM } from '../shared/realm-token';
 import { Token } from '../user/token';
+import * as jwt_decode from 'jwt-decode';
```

Comment:
Please move this up to line 4 so that it is with the other external imports.
Please specify exactly what you are importing as this impacts the size of the entire application. It can also introduce risk by including more external code to fail.